paasta_tools/autoscaling_lib.py

```diff
@@ -508,41 +512,130 @@ def spotfleet_scaler(resource, error):
 
     if new_capacity != current_capacity:
         print "Scaling SFR %s from %d to %d!" % (resource['id'], current_capacity, new_capacity)
-        scale_aws_spot_fleet_request(sfr_id=resource['id'], target_capacity=new_capacity, ec2_client=ec2_client)
+        return new_capacity - current_capacity, new_capacity
+    else:
+        return 0
+
+
+def slave_pid_to_ip(slave_pid):
+    """Convert slave_pid to IP
+
+    :param: slave pid e.g. slave(1)@10.40.31.172:5051
+    :returns: ip address"""
+    regex = re.compile(r'.+?@([\d\.]+):\d+')
+    return regex.match(slave_pid).group(1)
+
+
+def pick_slaves_to_kill(mesos_state, number, pool='default'):
```

Comment:
I think longer term this function has to have the actual list from the the SFR/ASG in order to work.

Otherwise it won't take into account instances that are not even registered in mesos (duds, broken puppet, etc)